list_test.py

```diff
@@ -0,0 +1,114 @@
+from list import List
+
+
+def create_list():
+    l = List()
+
+    l.push('p')
+    l.push('o')
+    l.push('n')
+    l.push('y')
+
+    return l
+
+
+def test_getitem_success():
+    assert create_list()[0] == 'p'
+
+def test_getitem_fail():
+    assert create_list()[0] != 'a'
+
+
+def test_setitem_success():
+    l1 = create_list()
+    l2 = create_list()
+
+    l1[-1] = 'k'
+    l2[-1] = 'k'
+
+    assert l1 == l2
+
+def test_setitem_fail():
+    l1 = create_list()
+    l2 = create_list()
+
+    l1[-1] = 'k'
+    l2[-1] = 't'
+
+    assert l1 != l2
+
+
+def test_equals_success():
+    l1 = create_list()
+    l2 = create_list()
+
+    assert l1 == l2
+
+def test_equals_fail_element():
+    l1 = create_list()
+    l1[0] = 'd'
+
+    assert l1 != create_list()
+
+def test_equals_fail_len():
+    l1 = create_list()
+    l1.remove(0)
+    
+    assert l1 != create_list()
+
+
+def test_push_success():
+    l1 = create_list()
+
+    l1.push('t')
+
+    assert len(l1) == len(create_list()) + 1 and l1[-1] == 't'
+
+def test_push_fail():
+    l1 = create_list()
+
+    l1.push('t')
+
+    assert len(l1) - 1 == len(create_list())
```

Comment:
Thank you! This structure that I needed. I assumed that there should be tests in which raising an exception is a success.

I have determined which methods require naming in `_success/_fail` tests: `push(), insert(), remove(), get and set`.
The other methods do not raise exceptions and it seems to me that they do not need `_success/_fail` tests.
#
I'm sorry, I don't understand what you want to tell me here:
```Python
# Note that it's not very important to do this assertion, because if `exception` is None
# then the assert after this one will also fail. If we do this assert separately it may just be easier
# to debug the test if it fails on `exception == None`. So feel free either to include this line of code or not
# assert exception != None
```
It seems to me that you meant that `exception` can be `None`? Like this:
```Python
actual_exception = None
try:
    create_list()[999] # Error!
except Exception as exception: # exception = None
    actual_exception = exception # then actual_exception = None

assert "Index 999 is out of bound" in str(actual_exception) # assert fails, 
# and there should be an `str(actual_exception)` here, not an `str(exception)`?
```